src/lib/io/ABC.cpp

```
@@ -0,0 +1,429 @@
+                    {
+                        Imath::V3f v(dataVec[i*3], dataVec[i*3+1], dataVec[i*3+2]);
+                        v = v * mat;
+                        dataVec[i*3] = v.x;
+                        dataVec[i*3+1] = v.y;
+                        dataVec[i*3+2] = v.z;
+                    }
+                }
+
+                for (std::size_t i=0; i < numPts; ++i)
+                {
+                    float * ptr = iData->dataWrite<float>(it->second, i);
+
+                    std::size_t idx = i * extent;
+                    // more points than we have data, so copy over the last
+                    // extent worth of data
+                    if (i >= dataPts)
+                    {
+                        idx = dataVec.size() - extent;
+                    }
+
+                    memcpy(ptr, &(dataVec[idx]), extent * sizeof(float));
+                }
+            }
+        }
+    }
+
+    for (std::size_t i = 0; i < iObj.getNumChildren(); ++i)
+    {
+        Abc::IObject child = iObj.getChild(i);
+        writePoints(child, iTime, iAttrMap, iData, oIndex, oMatStack);
+    }
+
+    if (shouldPop)
+    {
+        oMatStack.pop_back();
+    }
+}
+
+ParticlesDataMutable* readABC(const char* filename,
+                              const bool headersOnly,
+                              std::ostream* errorStream)
+{
+    // partio looks for the extension .abc to decide to call readABC
+    // and we want to be able to accept a time to read (right now in seconds)
+    // so we will pass in /tmp/foo@42.5.abc
+    std::string name = filename;
+    std::size_t lastAt = name.rfind('@');
+    double timeVal = -DBL_MAX;
+    if (lastAt != std::string::npos)
+    {
+        try
+        {
+            // we want to get everything FROM @ to but not including .abc
+            timeVal = stod(name.substr(lastAt + 1, name.size() - lastAt - 4));
+            name = name.substr(0, lastAt) + ".abc";
+        }
+        catch(...)
+        {}
```

We should be explicit about what we're catching here. It looks like the idea is that if it's not something that converts via `stod` then we'll ignore the exception it throws?